src/app/core/services/kafka.service.ts

```diff
@@ -38,6 +38,7 @@ export class KafkaService {
     private authService: AuthService
   ) {
     this.updateURI()
+    this.authService.refresh()
```

Comment:
Currently the authService refresh() function only updates the tokens if they are expired but we should update them if they are about to expire too (like if they are about to expire in the next 30 mins. This should be configurable in the default config) to avoid the scenario where the token was about to expire and was not updated at start of the app but after filling the questionnaire it expired and again results in problems with sending. What do you think ?